### ISSUE TITLE
March 2021 Endgame

### ISSUE THREAD

**Issue by TylerLeonhardt** [microsoft/vscode#119494]
- *3/22* Code freeze for the endgame
- *3/26* [Endgame done](https://github.com/Microsoft/vscode/wiki/Development-Process#inside-an-iteration)
- *4/03* Expected release date (this may change)

##### Monday
- [x] Run [OSS tool](https://github.com/microsoft/vscode-build-tools/tree/main/distro-tools) **endgame champion**
- [x] Create Test plan item for smoke testing ([template](https://github.com/microsoft/vscode/wiki/Test---Smoke-Test-Template)) **endgame champion**
- [x] Code freeze at 5pm PT
- [x] Ensure we have a green build on all platforms at 5pm PT
- [x] Create test plan items following the template [here](https://github.com/microsoft/vscode/wiki/Writing-Test-Plan-Items) by 6pm PT
- [x] Add verification-needed label to [features needed testing and are not tested by TPIs](https://github.com/microsoft/vscode/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3Afeature-request+is%3Aclosed+-label%3Aon-testplan+-label%3Averification-needed+).
- [x] Update your availability for testing here - https://vscode-tools.azurewebsites.net/testers

##### Tuesday
- [x] Test plan items assigned (using https://vscode-tools.azurewebsites.net/test-plan-items)
  - Run the tool multiple times to balance load if test items come in later and assignments are already made
  - [Assigned to you](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+assignee%3A%40me++label%3Atestplan-item+)
- [x] [🔖All closed feature-requests](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed+milestone%3A%22March+2021%22+label%3Afeature-request+-label%3Averification-needed+-label%3Aon-testplan+-label%3Averified+-label%3A*duplicate+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug) either have a `verification-needed` or `on-testplan` label
- [x] Test build starts at 7am CET
- [x] Test plan ready by 8am CET
- [x] [🔖Testing](https://github.com/issues?q=is%3Aopen+is%3Aissue+label%3Atestplan-item+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github+)
- [x] [🔖Verification needed](https://github.com/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22March+2021%22+label%3Averification-needed+-label%3Averified+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github)

##### Wednesday
- [x] [🔖Testing](https://github.com/issues?q=is%3Aopen+is%3Aissue+label%3Atestplan-item+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github+)
- [x] Remind team members to assign issues that they intend to fix to the current milestone
- [x] Fixing (self-assigned, milestone assigned)
- [x] [🔖Verification needed](https://github.com/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22March+2021%22+label%3Averification-needed+-label%3Averified+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github)
- [x] Prepare for Smoke test: Make sure all smoke test issues are labelled `smoke-test` and fixed **endgame champion**

##### Thursday
- [x] Fixing (self-assigned, milestone assigned, NO discussion during standup)
  - Move issues to the next month that can be deferred
- [x] [🔖Verification needed](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed+milestone%3A%22March+2021%22+label%3Averification-needed+-label%3Averified+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github)
- [x] [🔖Verification](https://github.com/issues?q=is%3Aclosed+is%3Aissue+milestone%3A%22March+2021%22+label%3Abug+-label%3Averified+-label%3Aon-testplan+-label%3Aduplicate+-label%3A*duplicate+-label%3Ainvalid+-label%3Aas-designed+-label%3Aerror-telemetry+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github+sort%3Aupdated-asc+)
- [x] Prepare for Smoke test: Make sure all smoke test issues are labelled `smoke-test` and fixed **endgame champion**

##### Friday
- [x] Pause scheduled `insider` builds **endgame champion**
- Satellite modules/npm packages ready, version updated, smoke tested
  - [x] vscode **@eamodio**
  - [x] yo generator **@aeschli**
  - [x] vsce **@joaomoreno**
  - [x] node debug **@weinand**
  - [x] node debug2 **@roblourens**
  - [x] js-debug **@connor4312**
  - [x] node debugadapter node **@weinand**
  - [x] debug adapter protocol **@weinand**
  - [x] custom data for html/css **@aeschli**
- [x] All issues [🔖verified](https://github.com/issues?q=is%3Aclosed+is%3Aissue+milestone%3A%22March+2021%22+label%3Abug+-label%3Averified+-label%3Aon-testplan+-label%3Aduplicate+-label%3A*duplicate+-label%3Ainvalid+-label%3Aas-designed+-label%3Aerror-telemetry+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug++repo%3Amicrosoft%2Fvscode-pull-request-github)
- [x] Fixing (scrutiny sets in - major bugs only - to be discussed in stand-up meeting, labeled as `candidate`)
- [Smoketest](https://github.com/Microsoft/vscode/wiki/Smoke-Test) (⚠️ MUST run with `--stable-build` argument ⚠️ )	
  - [x] Windows - **owner**	
  - [x] macOS (Universal) - **owner**	
  - [x] Linux - **owner**		
- [ ] All release notes updated
  - release notes are collected in a file named *`v<Major>_<Minor>.md`* in this [repo directory](https://github.com/Microsoft/vscode-docs/blob/vnext/release-notes/)
  - [x] @aeschli
  - [x] @alexdima
  - [x] @alexr00
  - [x] @bpasero
  - [x] @chrmarti
  - [x] @connor4312
  - [ ] @dbaeumer
  - [x] @deepak1556 
  - [x] @eamodio
  - [ ] @egamma
  - [x] @isidorn
  - [x] @JacksonKearl 
  - [x] @joaomoreno
  - [x] @jrieken
  - [ ] @kieferrm
  - [x] @lramos15
  - [ ] @lszomoru
  - [ ] @meganrogge
  - [x] @mjbvz
  - [x] @rebornix
  - [x] @rmacfarlane
  - [x] @roblourens
  - [x] @rzhao271
  - [ ] @sandy081
  - [ ] @sbatten
  - [x] @tylerleonhardt
  - [x] @tyriar
  - [x] @weinand
- [x] Acknowledge pull requests in release notes. We acknowledge PRs from outside the team. We have improved [the tooling](https://vscode-tools.azurewebsites.net/acknowledgement) so that the endgame champion can generate the pull request acknowledgment for all repositories at once. **@lszomoru**
  - [x] debug-adapter-protocol
  - [x] inno-updater
  - [x] jsonc-parser
  - [x] language-server-protocol
  - [ ] 💡localization - @weeteckt
  - [x] lsif-node
  - [x] vscode
  - [x] vscode-codicons
  - [x] vscode-css-languageservice
  - [x] vscode-debugadapter-node
  - [x] vscode-dev-containers
  - [x] vscode-docs
  - [x] vscode-emmet-helper
  - [x] vscode-eslint
  - [x] vscode-extension-samples
  - [x] vscode-generator-code
  - [x] vscode-hexeditor
  - [x] vscode-html-languageservice
  - [x] vscode-js-debug
  - [x] vscode-js-debug-companion
  - [x] vscode-js-profile-visualizer
  - [x] vscode-jshint
  - [x] vscode-json-languageservice
  - [x] vscode-languageserver-node
  - [x] vscode-loader
  - [x] vscode-lsif-extension
  - [x] vscode-node-debug
  - [x] vscode-node-debug2
  - [x] vscode-pull-request-github
  - [x] vscode-recipes
  - [x] vscode-references-view
  - [x] vscode-textmate
  - [x] vscode-vsce
- [x] Acknowledge [issue trackers](https://github.com/microsoft/vscode-internalbacklog/wiki/Community-Triage:-Credits) from the community **@chrmarti**
- [x] Add notable fixes to the release notes **all**
- When done fixing/verifying and there are changes since last build at the end of day PT
  - [x] Trigger new insider build and publish it manually **endgame champion**
- [x] Branch code to `release/<x.y>` **endgame champion**
- [x] Bump up the version in `package.json` on `main` - **endgame champion**
- [x] Announce `main` is open for business **endgame champion**

##### Friday/Monday
- [x] Polish release notes **redmond**
- [x] Fixing (only critical bugs - no string changes)

##### Monday - Wednesday
- [x] [🔖milestone issues](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+milestone%3A%22March+2021%22+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug)
- [x] [🔖candidate issues](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+label%3Acandidate+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug)
- [x] Polish release notes **redmond**
- [x] Cherry-pick hand-picked and reviewed changes to `release/<x.y>` **endgame champion**
- [x] Build `Insider` from `release/<x.y>` **endgame champion**
- [x] Manually release `Insider` **endgame champion**
- [x] Build stable for all platforms as new candidate issues come in **endgame champion**
- [x] Documentation updated
  - [x] @aeschli
  - [x] @alexdima
  - [ ] @alexr00
  - [x] @bpasero
  - [x] @chrmarti
  - [ ] @connor4312
  - [ ] @dbaeumer
  - [x] @deepak1556 
  - [x] @eamodio
  - [ ] @egamma
  - [x] @isidorn
  - [x] @JacksonKearl 
  - [x] @joaomoreno
  - [ ] @jrieken
  - [ ] @kieferrm
  - [x] @lramos15
  - [ ] @lszomoru
  - [x] @meganrogge
  - [x] @mjbvz
  - [x] @rebornix
  - [x] @rmacfarlane
  - [x] @roblourens
  - [x] @rzhao271
  - [ ] @sandy081
  - [ ] @sbatten
  - [x] @tylerleonhardt
  - [x] @tyriar
  - [x] @weinand
- [x] Run `scripts/test-documentation.sh|bat` and add file or fix issues if there are new colors that are not documented.

> **Note:** The `Insiders` build needs to be in the wild for 24 hours before we can enter the last phase of the endgame. **endgame champion**

##### Wednesday/Thursday - Expected release day (this may change)
- [x] Build stable for all platforms **endgame champion**
- [x] Sanity check of installable bits
  - [x] Windows 32 bit @roblourens 
    - [x] signed installer 32-bit
    - [x] signed user installer 32-bit
    - [x] zip 32-bit
  - [x] Windows 64 bit @TylerLeonhardt
    - [x] signed installer 64-bit
    - [x] signed user installer 64-bit
    - [x] zip 64-bit
  - [x] Windows ARM64 @eamodio
    - [x] signed installer ARM64
    - [x] signed user installer ARM64
    - [x] zip ARM64
  - [x] macOS @jrieken 
    - [x] Universal
    - [x] Intel
    - [x] Apple Silicon
  - [x] Linux x64 @RMacfarlane 
    - [x] deb
    - [x] rpm
    - [x] archives
    - [x] snap (`sudo snap install --classic --dangerous <file>.snap`)
  - [x] Server ([instructions](https://github.com/microsoft/vscode-remote-release/wiki/Sanity-Check-VS-Code-Servers)) **owner**
    - [x] Linux @TylerLeonhardt 
    - [x] macOS @jrieken 
    - [x] Windows @TylerLeonhardt 
- [x] Publish website **@ornellaalt**
- [ ] Publish Localization language pack **@weeteckt**
- [x] Publish to stable **endgame champion**
- [x] Create an official release **endgame champion**
  - [x] Create a tag: `git tag <x.y.z>`
  - [x] Push the tag: `git push origin <x.y.z>`
  - [x] Create a GitHub release: [Open the GitHub tags](https://github.com/microsoft/vscode/tags), and click far right `... > Create Release`. Use the correct title and description from our release notes. [Example](https://github.com/microsoft/vscode/releases/tag/1.39.0)
- [x] [Publish @types/vscode](https://github.com/microsoft/vscode/wiki/Publish-vscode-types) **endgame champion**
- [x] Enable scheduled `insider` builds **endgame champion**
- [x] Twitter announcement **@chrisdias**

**Comment by yubaoquan** [comment]
> 3/03 Expected release date

not 4/03 ?

**Comment by bugpowder** [comment]
That was fast!
